api.ts

```diff
@@ -86,7 +85,7 @@ export interface CustomConfigurationProvider extends vscode.Disposable {
      * @param token (optional) The cancellation token.
      * @returns 'true' if this provider can provide IntelliSense configurations for the given document.
      */
-    canProvideConfiguration(uri: vscode.Uri, token?: CancellationToken): Thenable<boolean>;
+    canProvideConfiguration(uri: vscode.Uri, token?: vscode.CancellationToken): Thenable<boolean>;

```

Comment:
Are the interfaces the same?  If not, then we can't change this without fixing cpptools and updating the major revision number of the API.

Reply:
You also need to update `api.d.ts` in the `out` folder.  IIRC, it should happen automatically if you npm compile this project.

Reply:
> Are the interfaces the same? If not, then we can't change this without fixing cpptools and updating the major revision number of the API.

The interface is the same.
![image](https://user-images.githubusercontent.com/19859882/103689915-cd90f180-4f48-11eb-906f-1cae2e1fdc20.png)

Reply:
> You also need to update `api.d.ts` in the `out` folder. IIRC, it should happen automatically if you npm compile this project.

Oh, I broke the compile script because I removed ./node_modules/vscode/bin/install . Looking for a replacement...

Reply:
It looks like we're removing restrictions `readonly` which should not require a major version update.  If it was the other way around (e.g. adding `readonly`), we would have to update the major version.  But I think this is all moot since @michelleangela is going to be changing intelliSenseMode which will require an update anyway.

Reply:
I'm pretty sure the major version doesn't need to change over CancellationToken, because our vscode-cpptools repo was already using a vscode.CancellationToken type instead of the vscode-jsonrpc one, and even if another repo was using the vscode-jsonrpc type, the change to from read only to not read only wouldn't be a breaking change since it's going from more restrictive to less restrictive.

Reply:
> and even if another repo was using the vscode-jsonrpc type, the change to from read only to not read only wouldn't be a breaking change since it's going from more restrictive to less restrictive.

Yes, I think that's what I said.

Reply:
Yeah, I wrote my comment before I saw your post.